src/components/Progress/ProgressLine.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { ReactNode, forwardRef } from 'react'
+import { composeClasses } from 'lib/classes'
+
+interface ProgressLineProps extends React.HTMLAttributes<HTMLDivElement> {
```

Comment:
You repetead all base props on all interfaces, `ProgressCircleProps` and `ProgressProps`.
We can create the base props interface and extends it in each one